lib/core-server/src/utils/stories-json.ts

```diff
@@ -15,21 +19,51 @@ export async function extractStoriesJson(
 }
 
 export async function useStoriesJson(router: any, options: Options) {
-  const normalized = normalizeStories(await options.presets.apply('stories'), {
+  const normalizedStories = normalizeStories(await options.presets.apply('stories'), {
     configDir: options.configDir,
     workingDir: process.cwd(),
   });
+  const generator = new StoryIndexGenerator(normalizedStories, options.configDir);
+  await generator.initialize();
+
+  const invalidationEmitter = new EventEmitter();
+  watchStorySpecifiers(normalizedStories, (specifier, path, removed) => {
+    generator.invalidate(specifier, path, removed);
+    console.log('emitting');
+    invalidationEmitter.emit(eventName);
+  });
 
-  router.use('/stories.json', async (_req: any, res: any) => {
-    const generator = new StoryIndexGenerator(normalized, options.configDir);
-    await generator.initialize();
+  router.use('/stories.json', async (req: any, res: any) => {
+    if (req.headers.accept === 'text/event-stream') {
+      let closed = false;
+      const watcher = () => {
+        if (closed || res.writableEnded) return;
+        res.write(`event:INVALIDATE\ndata:DATA\n\n`);
```

Comment:
```suggestion
        res.write(`event:${INVALIDATE}\ndata:DATA\n\n`);
```

Reply:
what's `DATA`?

Reply:
Oh yeah whoops. Thanks. Took my a while to get this to work